wgpu-core/src/error.rs

```diff
@@ -0,0 +1,321 @@
+            Self::InvalidSampler(id) => {
+                let name = crate::gfx_select!(id => global.sampler_label(id));
+                ret.push_str(&format_label_line("sampler", &name));
+            }
+            _ => {}
+        };
+        ret
+    }
+}
+
+impl PrettyError for crate::binding_model::CreatePipelineLayoutError {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        let mut ret = format_error_line(self);
+        if let Self::InvalidBindGroupLayout(id) = *self {
+            let name = crate::gfx_select!(id => global.bind_group_layout_label(id));
+            ret.push_str(&format_label_line("bind group layout", &name));
+        };
+        ret
+    }
+}
+
+impl PrettyError for crate::command::ExecutionError {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        let mut ret = format_error_line(self);
+        match *self {
+            Self::DestroyedBuffer(id) => {
+                let name = crate::gfx_select!(id => global.buffer_label(id));
+                ret.push_str(&format_label_line("buffer", &name));
+            }
+            Self::Unimplemented(_reason) => {}
+        };
+        ret
+    }
+}
+
+impl PrettyError for crate::command::RenderPassErrorInner {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        let mut ret = format_error_line(self);
+        if let Self::InvalidAttachment(id) = *self {
+            let name = crate::gfx_select!(id => global.texture_view_label(id));
+            ret.push_str(&format_label_line("attachment", &name));
+        };
+        ret
+    }
+}
+
+impl PrettyError for crate::command::RenderPassError {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        // This error is wrapper for the inner error,
+        // but the scope has useful labels
+        format_error_line(self) + &self.scope.fmt_pretty(global)
+    }
+}
+
+impl PrettyError for crate::command::ComputePassError {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        // This error is wrapper for the inner error,
+        // but the scope has useful labels
+        format_error_line(self) + &self.scope.fmt_pretty(global)
+    }
+}
+impl PrettyError for crate::command::RenderBundleError {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        // This error is wrapper for the inner error,
+        // but the scope has useful labels
+        format_error_line(self) + &self.scope.fmt_pretty(global)
+    }
+}
+
+impl PrettyError for crate::command::ComputePassErrorInner {
+    fn fmt_pretty(&self, global: &Global<IdentityManagerFactory>) -> String {
+        let mut ret = format_error_line(self);
+        match *self {
+            Self::InvalidBindGroup(id) => {
+                let name = crate::gfx_select!(id => global.bind_group_label(id));
+                ret.push_str(&format_label_line("bind group", &name));
```

Comment:
this should probably look like `format_label_line(&mut ret, "bind group", &name)`, so that it writes to the string directly instead of first creating a new string and then taking a reference of it

Reply:
Hmm yeah, if the `fmt::Formatter` etc. changes don't pan out, then that seems like a worthwhile change.